core/src/main/java/com/google/googlejavaformat/java/Formatter.java

```diff
@@ -151,7 +151,17 @@ public CharSequence getCharContent(boolean ignoreEncodingErrors) throws IOExcept
     OpsBuilder builder = new OpsBuilder(javaInput, javaOutput);
     // Output the compilation unit.
     JavaInputAstVisitor visitor;
-    if (Runtime.version().feature() >= 17) {
+    if (Runtime.version().feature() >= 21) {
+      try {
+        visitor =
+            Class.forName("com.google.googlejavaformat.java.java21.Java21InputAstVisitor")
+                .asSubclass(JavaInputAstVisitor.class)
+                .getConstructor(OpsBuilder.class, int.class)
+                .newInstance(builder, options.indentationMultiplier());
```

Comment:
It might be worth creating a shared helper to reflectively instantiate the visitor at this point:

```
private static JavaInputAstVisitor createVisitor(String className) { .. }
```